b2sdk/transfer/inbound/downloaded_file.py

```diff
@@ -93,12 +108,21 @@ def __enter__(self):
 
         # All remaining problems should be with permissions.
         try:
-            self.file = open(self.path_, self.mode, buffering=self.buffering)
+            self.file = open(
+                self.path_,
+                'wb+' if self.mode is None else self.mode,
+                buffering=self.buffering,
+            )
+        except io.UnsupportedOperation:
+            if self.mode is not None:
+                raise
```

Comment:
if someone asked to open the file in a specific mode and that can't be done, error out. You shouldn't be trying to repair it as it's unexpected behavior, someone may be running this on top of a fuse which does something funny on open(). Fall back to 'wb' only if `mode is None`